### PR TITLE
Feature/dssm inte test

### PR DESCRIPTION
**Pull request by bwanglzu**
For issue #245 
1. add integration test for dssm.
2. Resolve bug in dssm_preprocessor
3. Add utility.
4. minor changes.

### CODE REVIEW

File: tests/inte_test/test_dssm.py [hunk 1]
```diff
@@ -26,11 +30,30 @@ def inte_test_dssm():
     # do pre-processing.
     dssm_preprocessor = preprocessor.DSSMPreprocessor()
```
Comment (faneshion) [inline tests/inte_test/test_dssm.py]: Here, we also need to save `dssm_preprocessor ` for prediction phase. Thus, we can load `dssm_preprocessor` for testing.

File: tests/inte_test/test_dssm.py [hunk 2]
```diff
@@ -26,11 +30,30 @@ def inte_test_dssm():
...
+    dssm_model = models.DSSMModel()
+    dssm_model.params['input_shapes'] = [(dim_triletter, ), (dim_triletter, )]
+    dssm_model.guess_and_fill_missing_params()
+    dssm_model.build()
+    dssm_model.compile()
+    dssm_model.fit([X['text_left'], X['text_right']], y)
+    dssm_model.save('.tmpdir')
+
+    # testing
+    processed_test = dssm_preprocessor.fit_transform(test, stage='test')
```
Comment (faneshion) [inline tests/inte_test/test_dssm.py]: Here,  the testing phase can be implemented in an separate test function. Thus, we can also test `load dssm_preprocessors`.

File: tests/inte_test/test_dssm.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    processed_train = dssm_preprocessor.fit_transform(train, stage='train')
+    # the dimension of dssm model is the length of tri-letters.
+    dim_triletter = processed_train.context['dim_triletter']
+    # generator.
+    generator = generators.PointGenerator(processed_train)
+    X, y = generator[0]
+    # Create a dssm model
+    dssm_model = models.DSSMModel()
+    dssm_model.params['input_shapes'] = [(dim_triletter, ), (dim_triletter, )]
+    dssm_model.guess_and_fill_missing_params()
```
Comment (pl8787) [inline tests/inte_test/test_dssm.py]: Does `guess_and_fill_missing_params` function must be called explicitly?
I think it should be included in `build` function.

Comment (bwanglzu) [reply tests/inte_test/test_dssm.py]: @pl8787  current needed, in the future we'll write a function as "facade" to wrap these building blocks.

**Pull request merged**